liteflow-labs/liteflow-js
# Remove session from components

### Project organization

- Related https://github.com/liteflow-labs/liteflow-js/issues/26
- Related https://github.com/liteflow-labs/marketplace-template/issues/46

### Description

Remove the session component and the need for most of the `LiteflowNFTApp` contexts.

This PR can/should be reviewed in parallel with https://github.com/liteflow-labs/marketplace-template/pull/44 which implements all the changes following similar commit history. Also, checking this PR commit per commit will be much easier.

Things done:
- Move useEagerConnect to app and delete from Session
- Delete connectors from session
- Remove cookies from session.tsx
- Remove signer from session
- Move authenticate wallet to App by using activate/deactivate from web3 react, similar to minimal app
- Replace useSession() calls with useWeb3React equivalent data
- Delete the Session component and context

### How to test

With https://github.com/liteflow-labs/marketplace-template/pull/44 you will need to build the libraries and pack them to install them with no issue in the template
- In this repo: `npm pack -ws`
- In the marketplace template PR: `npm i ../liteflow-js/*.tgz`

Then you'll need to make sure to test the connection to wallet, switch between accounts, reconnection when page is reloaded etc...
You can also check if requests are correctly sent to the API with the proper authentication token, the best way is to use an NFT with locked content so only the owner will be able to retrieve the right information on the API.

### Checklist
